### ISSUE TITLE
Add examples with DI frameworks

### ISSUE THREAD

**Issue by filiphr** [mapstruct/mapstruct-examples#24]
Add examples with DI frameworks:

* [ ] - Spring
* [ ] - Guice
* [ ] - Dagger 2
* [ ] - CDI

**Comment by xenteros** [comment]
As far as I know, MapStruct mappers are Beans by default, so you can easily @Autowire them. Did you have any difficulties with autowiring mappers?

**Comment by xenteros** [comment]
I can prepare Spring examples if needed. I will, when I'll have some time.

**Comment by filiphr** [comment]
That would be great @xenteros. I think that it is good to have as many examples as possible so people can see how exactly they can configure their project. We have an integration test with Spring as well, but not with the other DI frameworks

**Comment by xenteros** [comment]
@filiphr I have just created pull request with example that was requested on StackOverflow from time to time. Would you have a look?

**Comment by filiphr** [comment]
@xenteros yes I've seen your PR. I didn't had time to have a better look. I'll try and do it by the end of the weekend, I hope that is fine for you.

**Comment by xenteros** [comment]
@filiphr yup. I'll try to find more interesting examples in the code I have already wrote and rewrite, so they fit examples.
